doc-content/enterprise-only/project-data/variables-con.adoc

```diff
@@ -19,6 +19,63 @@ If the variable cannot be found directly in the element's context, the immediate
 If the variable cannot be found, a read access request returns `null` and a write access produces an error message, and the process continues its execution.
 Variables are searched for based on their ID.
 
+== Variable tags
+For greater control over variable behavior, you can tag process variables and local variables in the BPMN process file. Tags are simple string values that you add as metadata to a specific variable.
+
+{PRODUCT} supports the following tags for process variables and local variables:
+
+* `required`: Sets the variable as a requirement in order to start a process instance. If a process instance starts without the required variable, {PRODUCT} generates a `VariableViolationException` error.
+* `readonly`: Indicates that the variable is for informational purposes only and can be set only once during process instance execution. If the value of a read-only variable is modified at any time, {PRODUCT} generates a `VariableViolationException` error.
+* `restricted`: Indicates a special tag that is used with `VariableGuardProcessEventListener` for granted permission to modify the variable based on the required and the existing role.
++
+`VariableGuardProcessEventListener` is extended from `DefaultProcessEventListener` and supports two different constructors:
++
+[source,java]
+----
+public VariableGuardProcessEventListener(String requiredRole, IdentityProvider identityProvider) {
```

Comment:
I would separate these with bullets.
* `VariableGuardProcessEventListener`
[source]
----
public VariableGuardProcessEventListener(String requiredRole, IdentityProvider identityProvider) {
    this("restricted", requiredRole, identityProvider);
}
* `VariableGuardProcessEventListener`
[source]
----
public VariableGuardProcessEventListener(String tag, String requiredRole, IdentityProvider identityProvider) {
    this.tag = tag;
    this.requiredRole = requiredRole;
    this.identityProvider = identityProvider;
}
----

Reply:
Hard to see here but edit the comment for the code.

Reply:
Done